lib/plugins.ts

```diff
@@ -508,7 +509,10 @@ export class Plugins {
               .map(
                   pluginObj =>
                       this.safeCallPluginFun(pluginObj, funName, args, promiseType, failReturnVal));
-      return promiseType == PromiseType.Q ? q.all(promises) : webdriver.promise.all(promises);
+      return promiseType == PromiseType.Q ?
```

Comment:
We only use this designation internally, and the expectations for method support would all be internal though right? As long as the plugin author returns a `then`able we should be able to resolve it with `Promise.all` in Runner/Browser. I could have totally missed something here though, so let me know if I haven't gotten it right 😄 

If there's no contract with `Q` i'd love to just do away with this branching and make a WD promise the only special case.